File: src/main/java/com/codeperfection/shipit/dto/shipping/ShippingDto.java
package com.codeperfection.shipit.dto.shipping;

import com.codeperfection.shipit.dto.transporter.TransporterDto;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;

import java.time.OffsetDateTime;
import java.util.List;
import java.util.UUID;

@Data
@Builder
@AllArgsConstructor
public class ShippingDto {

    private UUID uuid;

    private String name;

    private OffsetDateTime createdAt;

    private TransporterDto transporter;

    private List<ShippedItemDto> shippedItems;
}

File: src/main/kotlin/com/codeperfection/shipit/dto/shipping/ShippingDto.kt
package com.codeperfection.shipit.dto.shipping

import com.codeperfection.shipit.dto.transporter.TransporterDto
import com.codeperfection.shipit.entity.Shipping
import java.time.OffsetDateTime
import java.util.*

data class ShippingDto(
    val id: UUID,
    val name: String,
    val userId: UUID,
    val createdAt: OffsetDateTime,
    val transporter: TransporterDto,
    val shippedItems: List<ShippedItemDto>
) {

    companion object {

        fun fromEntity(shipping: Shipping) = ShippingDto(
            id = shipping.id,
            name = shipping.name,
            createdAt = shipping.createdAt,
            userId = shipping.userId,
            transporter = TransporterDto.fromEntity(shipping.transporter),
            shippedItems = shipping.shippedItems.map(ShippedItemDto::fromEntity)
        )
    }
}
